### PR TITLE
Add osx build

### PR DESCRIPTION
**Pull request by jdblischak**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

* Follow up to #4 
* The linux build is failing, so this only adds the osx build
* @jjerphan - please comment below if you are willing to help maintain this recipe

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -10,26 +10,29 @@ source:
...
   build:
-    - vs2017_win-64          # [win64]
+    - {{ compiler('c') }}
     - cmake
-    - ninja
+    - make   # [not win]
+    - ninja  # [win]
   host:
-    - azure-macro-utils-c
+    - azure-macro-utils-c <2020.10.29
```
Comment (jjerphan) [inline recipe/meta.yaml:25]: Could you indicate why pinning is needed here?

File: recipe/meta.yaml [hunk 2]
```diff
@@ -10,26 +10,29 @@ source:
     sha256: 643f3b84ac511b301e481c35278e6a4deacb9e945ad1e36dc3ff4b5f45300b39
 
 build:
-  number: 0
-  skip: true  # [not win64]
+  number: 1
+  skip: true  # [linux]
```
Comment (jjerphan) [inline recipe/meta.yaml:14]: Do you know if there is something blocking Linux for now or not? If not, I can open another PR for it once this one is merged.

Comment (jdblischak) [reply recipe/meta.yaml:14]: The blocker of the linux build is that openssl error I shared with you yesterday in https://github.com/conda-forge/azure-c-shared-utility-feedstock/pull/4#issuecomment-1668180004

Comment (jdblischak) [reply recipe/meta.yaml:14]: I was going to open the PR for the linux build, but if you're eager to do it, then I'm happy to let you have at it

File: recipe/meta.yaml (hunk 1 above)
Comment (jdblischak) [reply recipe/meta.yaml:25]: The library name changed

https://github.com/conda-forge/umock-c-feedstock/pull/3/files#r948327329

Comment (jjerphan) [reply recipe/meta.yaml:25]:
```suggestion
    # 2020.10.29 is the first incompatible version after
    # azure-macro-utils-c was renamed to macro-utils-c.
    - azure-macro-utils-c <2020.10.29
```

Comment (jdblischak) [reply recipe/meta.yaml:28]: Commit from 2020-10-29 that changed the name: https://github.com/Azure/macro-utils-c/commit/39be380162040dc8df723105c857552f6279e8da

**Review by xylar — APPROVED** [review]
This looks good to me!  Thanks for the work!

**Comment by jjerphan** [comment]
Thank you for approving this PR, @jdblischak.

It does not seem that the `automerge` was performed. Must another condition be verified?

**Comment by xylar** [comment]
Not sure sure why the automerge didn't work. Thanks for catching that.

**Pull request merged**